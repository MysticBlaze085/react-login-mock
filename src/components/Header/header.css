.header {
  display: flex;
  align-items: center;
  height: 50px;
  padding: 0px 40px;
  box-shadow: 2px 2px 4px #c5c3c3;
}

.logo-header {
  height: 20px;
}
