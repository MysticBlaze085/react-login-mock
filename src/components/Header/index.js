import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

import logoSrc from "../../assets/logo.svg";
import "./header.css";

function Header({ title }) {
  const [isLogged, setIsLogged] = useState(false);

  useEffect(() => {
    const authState = window.localStorage.getItem("@test:auth");
    if (authState) {
      const { isLogged } = JSON.parse(authState);
      if (isLogged) setIsLogged(true);
    }
  }, []);

  const handleLogout = () => {
    window.localStorage.removeItem("@test:auth");
  };

  return (
    <header className="header">
      <div className="logo-view">
        <Link to="/">
          <img className="logo-header" src={logoSrc} alt="Dev World Logo" />
        </Link>
        <p data-testid="header-title" className="title">
          {title}
        </p>
      </div>
      <menu>
        <Link className="link" to="/about">
          About
        </Link>
        {isLogged ? (
          <>
            <Link className="link" to="/profile">
              Profile
            </Link>
            <Link className="link" to="/login" onClick={handleLogout}>
              Logout
            </Link>
          </>
        ) : (
          <Link className="link" to="/login">
            Login
          </Link>
        )}
      </menu>
    </header>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
};

Header.defaultProps = {
  title: "",
};

export default Header;
