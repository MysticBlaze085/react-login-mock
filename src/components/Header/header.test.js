import React from "react";
import { render } from "@testing-library/react";
import Header from "./index";

test("renders header component", () => {
  const { asFragment, getByAltText, getByText, getByTestId } = render(
    <Header title="Title" />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(getByAltText(/logo/i)).toBeInTheDocument();
  expect(getByText("Title")).toBeInTheDocument();
  // expect(getByTestId("header-title")).toBeInTheDocument();
});
