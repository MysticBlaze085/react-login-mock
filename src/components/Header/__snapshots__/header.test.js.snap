// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header component 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <img
      alt="Healthify Logo"
      class="logo-header"
      src="logo.png"
    />
    <p
      class="title"
      data-testid="header-title"
    >
      Title
    </p>
  </header>
</DocumentFragment>
`;
